lib/kubernetes-deploy/pod_logger.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class PodLogger
+    def initialize(stream:, logger:, pod_name:, pod_id:)
+      @stream = stream
+      @logger = logger
+      @pod_name = pod_name
+      @pod_id = pod_id
+    end
+
+    def log(kubectl:, deploy_succeeded:, deploy_started_at:, unmanaged:, container_names:)
```

Comment:
It is intentionally not available in `Pod#new` (where this is currently initialized) because it is really important that `kubectl` not be used outside of `sync` (which is given the kubectl instance) in the resource classes.

Reply:
For performance reasons we only allow requests to the k8s api servers through the sync method. Adding this to the initializer would make it easy to violate that constraint.